jenkins/jobs/dsl/docker_pipeline_jobs.groovy

```diff
@@ -357,12 +300,74 @@ imageTest.with{
             | cat ${WORKSPACE}/${JOB_NAME##*/}.out
             |fi
             |
-            |echo LOGIN=$(echo ${DOCKER_LOGIN}) > credential.properties
             |set +x'''.stripMargin())
-    environmentVariables {
-      propertiesFile('credential.properties')
+  }
+  publishers{
+    downstreamParameterized{
+      trigger(projectFolderName + "/BDD_Security_Test"){
+        condition("UNSTABLE_OR_BETTER")
+        parameters{
+          predefinedProp("B",'${B}')
+          predefinedProp("PARENT_BUILD", '${PARENT_BUILD}')
+          predefinedProp("IMAGE_TAG",'${IMAGE_TAG}')
+        }
+      }
+    }
+  }
+}
+
+bddSecurityTest.with{
+  description("This job uses a docker-security-test a reusable container security testing tool that is designed to look for common security errors which have been defined using a BDD approach.")
+  parameters{
+    stringParam("B",'',"Parent build number")
+    stringParam("PARENT_BUILD","Get_Dockerfile","Parent build name")
+    stringParam("IMAGE_TAG",'',"Enter a unique string to tag your images e.g. your enterprise ID (Note: Upper case chararacters are not allowed)")
+  }
+  wrappers {
+    preBuildCleanup()
+    injectPasswords()
+    maskPasswords()
+    sshAgent("adop-jenkins-master")
+  }
+  environmentVariables {
+      env('WORKSPACE_NAME',workspaceFolderName)
+      env('PROJECT_NAME',projectFolderName)
+  }
+  label("docker")
+  steps {
+    copyArtifacts("Get_Dockerfile") {
+      buildSelector {
+          buildNumber('${B}')
+      }
     }
+    shell('''set -xe
+            |echo "Use the darrenajackson/image-inspector container to inspect the image"
+            |# Set path workspace is available from inside docker machine
+            |export docker_workspace_dir=$(echo ${WORKSPACE} | sed 's#/workspace#/var/lib/docker/volumes/jenkins_slave_home/_data#')
+            |docker run --rm \\
```

Comment:
Hi @anton-kasperovich, @NinjaPT's luismsousa/docker-security-test image gets cleaned up by the --rm as you'd expect, but it also creates it's own containers as part of the testing and we struggled get it to clear these up properly (during the test tear downs).  So the explicit rm line is a bit of a hack to tidy up properly.